### PR TITLE
[merge after #1208][jsk_fetch_startup] add jsk_fetch.urdf.xacro in jsk_fetch_startup

### PR DESCRIPTION
**Pull request by knorth55**
depends on #1199, #1200, #1208 and #1250

add `jsk_fetch.urdf.xacro` for Fetch in JSK, which has additional camera mount and head box.

fetch_urdf

cc. @sktometometo @708yamaguchi

### CODE REVIEW

**Comment by knorth55** [comment]
@sktometometo can you fix the nodelet problems?

**Comment by sktometometo** [comment]
@knorth55 I'll try.

**Comment by knorth55** [comment]
@sktometometo 
```log
[FATAL] [1582297391.527054925] [/head_camera/rgb/throttle_rgb:ros.nodelet]: Failed to load nodelet '/head_camera/rgb/throttle_rgb` of type `jsk_topic_tools/LightweightThrottle` to manager `/head_camera/he
ad_camera_nodelet_manager'
[FATAL] [1582297391.527281329] [/head_camera/rgb/downsample_half:ros.nodelet]: Failed to load nodelet '/head_camera/rgb/downsample_half` of type `image_proc/resize` to manager `/head_camera/head_camera_no
delet_manager'
[FATAL] [1582297391.527165786] [/head_camera/rgb/throttle_camera_info:ros.nodelet]: Failed to load nodelet '/head_camera/rgb/throttle_camera_info` of type `jsk_topic_tools/LightweightThrottle` to manager 
`/head_camera/head_camera_nodelet_manager'
[FATAL] [1582297391.528868160] [/head_camera/depth_registered/throttle_image_compressed:ros.nodelet]: Failed to load nodelet '/head_camera/depth_registered/throttle_image_compressed` of type `jsk_topic_to
ols/LightweightThrottle` to manager `/head_camera/head_camera_nodelet_manager'
[FATAL] [1582297391.528925017] [/head_camera/depth_registered/throttle_image:ros.nodelet]: Failed to load nodelet '/head_camera/depth_registered/throttle_image` of type `jsk_topic_tools/LightweightThrottl
e` to manager `/head_camera/head_camera_nodelet_manager'
[FATAL] [1582297391.529110899] [/head_camera/depth_registered/throttle_cloud:ros.nodelet]: Failed to load nodelet '/head_camera/depth_registered/throttle_cloud` of type `jsk_topic_tools/LightweightThrottl
e` to manager `/head_camera/head_camera_nodelet_manager'
[FATAL] [1582297391.529166851] [/head_camera/depth_registered/relay_image_compressed:ros.nodelet]: Failed to load nodelet '/head_camera/depth_registered/relay_image_compressed` of type `jsk_topic_tools/Re
lay` to manager `/head_camera/head_camera_nodelet_manager'
[FATAL] [1582297391.529244941] [/head_camera/depth_registered/resize_cloud:ros.nodelet]: Failed to load nodelet '/head_camera/depth_registered/resize_cloud` of type `jsk_pcl/ResizePointsPublisher` to mana
ger `/head_camera/head_camera_nodelet_manager'
[FATAL] [1582297391.528972403] [/head_camera/depth_registered/throttle_camera_info:ros.nodelet]: Failed to load nodelet '/head_camera/depth_registered/throttle_camera_info` of type `jsk_topic_tools/Lightw
eightThrottle` to manager `/head_camera/head_camera_nodelet_manager'
[FATAL] [1582297391.528979507] [/head_camera/rgb/downsample_quater:ros.nodelet]: Failed to load nodelet '/head_camera/rgb/downsample_quater` of type `image_proc/resize` to manager `/head_camera/head_camer
a_nodelet_manager'
[FATAL] [1582297391.529756947] [/head_camera/depth_registered/relay_image:ros.nodelet]: Failed to load nodelet '/head_camera/depth_registered/relay_image` of type `jsk_topic_tools/Relay` to manager `/head
_camera/head_camera_nodelet_manager'
```

**Comment by 708yamaguchi** [comment]
Currently, `jsk_fetch.urdf.xacro` is loaded to `/robot_description` in `/etc/ros/indigo/robot.launch`, which only exists on fetch PC.

If we use JSK's own xacro, should `jsk_fetch.urdf.xacro` be loaded in `jsk_fetch_startup/launch/fetch_bringup.launch` instead of `/etc/ros/indigo/robot.launch`?

File: jsk_fetch_robot/jsk_fetch_startup/robots/jsk_fetch.urdf.xacro [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" ?>
+<robot name="fetch" xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:include filename="/etc/ros/indigo/calibrated_latest.urdf" />
```
Comment (knorth55) [inline jsk_fetch_robot/jsk_fetch_startup/robots/jsk_fetch.urdf.xacro:3]: @708yamaguchi we include this file, and we made symbolic link for the latest calibrated urdf.

**Comment by knorth55** [comment]
@708yamaguchi see this comment. https://github.com/jsk-ros-pkg/jsk_robot/pull/1205#discussion_r442781525
first we make symbolic link to the current urdf to `/etc/ros/indigo/calibrated_latest.urdf` and include it in `jsk_fetch.urdf.xacro`.

**Comment by 708yamaguchi** [comment]
@knorth55 
Thank you for the explanation.
I was concerned that `/etc/ros/indigo/robot.launch` has a link to `jsk_fetch_startup` (JSK's own package)

In my understanding, 
- `/etc/ros/indigo/robot.launch` is to call programs which is independent of JSK packages
- `jsk_fetch_startup/launch/fetch_bringup.launch` is to call programs which is in JSK packages

What do you think the role of these files is?

**Comment by knorth55** [comment]
@708yamaguchi that is ideal, but we will modify the robot hardware, so we need to modify `robot.launch`.
or we can overwrite the parameter in `fetch_bringup.launch` for JSK fetch, and keep `robot.launch` independent from `jsk` packages.
(however, we switch to use `robot_localization` now, and fetch `odom` `imu` topic is incorrect, so we made `imu_corrector.py` in jsk packges, which means `robot.launch` already depends on `jsk` packages)

**Comment by 708yamaguchi** [comment]
Thank you for your explanation again.
I understand your thoughts.

I hope that `robot.launch` will be merged into `jsk_fetch_startup` soon.
https://github.com/jsk-ros-pkg/jsk_robot/pull/1208

**Comment by knorth55** [comment]
Now, with this PR, you can get fetch with base camera and head box.

fetch_urdf

cc. @708yamaguchi @sktometometo

**Comment by knorth55** [comment]
move to #1383

**Pull request closed without merging**